api/router.go

```diff
@@ -59,20 +64,21 @@ func noCacheMiddleware(next http.Handler, dt Dt) http.Handler {
 		dt.RunPullerChan <- true
 		select {
 		case <-dt.RunPullerDoneChan:
-			log.Debugf("Successfully collected cluster health status")
-			break
+			log.Debug("Fresh data updated")
+
 		case <-time.After(time.Minute):
-			e := "Puller timeout occured"
-			log.Error(e)
-			http.Error(w, e, http.StatusRequestTimeout)
-			return
+			panic("Error getting fresh health report")
```

Comment:
this is intensional, we are trying to prevent a deadlock here. If for some reason we did not get response to `RunPullerDoneChan` that means something stuck while pulling the health report. It is better to panic and let systemd restart the service, rather then be in weird stuck state.